Libplanet.Tests/Net/SwarmTest.cs

```diff
@@ -2388,6 +2388,40 @@ public async Task CreateNewChainWhenBranchPointNotExist()
             }
         }
 
+        [Fact(Timeout = Timeout)]
+        public async Task DoNotRemoveCanonicalChainWhenBlockDownloadFailed()
+        {
+            var chainA = _blockchains[0];
+            var chainB = _blockchains[1];
+
+            var swarmA = _swarms[0];
+            var swarmB = _swarms[1];
+
+            var genesis = await chainA.MineBlock(_fx1.Address1);
+            chainB.Append(genesis);
+
+            var block = await chainA.MineBlock(_fx1.Address1);
+
+            try
+            {
+                await StartAsync(swarmA);
+                await StartAsync(swarmB);
+                await BootstrapAsync(swarmA, swarmB.AsPeer);
+
+                swarmA.BroadcastBlocks(new[] { block });
+                await Task.Delay(100);
```

Comment:
This scenario seems likely to make the test flaky, because `swarmA.BroadcastBlocks(new[] { block })` could be faster than we assume… how about watching `chainB.TipChanged` event instead?

Reply:
Unfortunately, we can't use the `TipChanged` event because block downloading should be failed before block appending is complete.

Reply:
Could we make a new event to test this then?

Reply:
I added `Swarm<T>. FillBlocksAsyncFailed` for this test case.